internal/util/password/scramsha256_test.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package password
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/xdg-go/scram"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
+)
+
+// scramSHA256TestCase represents a test case for SCRAM-SHA-256 authentication.
+//
+//nolint:vet // for readability
+type scramSHA256TestCase struct {
+	params   scramSHA256Params
+	username string
+	password string
+	salt     []byte
+
+	want *types.Document
+	err  string
+}
+
+var scramSHA256TestCases = map[string]scramSHA256TestCase{
```

Comment:
What is the source of those test cases?

Reply:
I've tried to find SCRAM-SHA-256 test vectors but haven't found any source (e.g., NIST Cryptographic Algorithm Validation Program, PostgreSQL, and other open-source code) with precise input/output values I could use to recalculate and match against our implementation.

So I added test cases with values generated by MongoDB and explained how to generate such values in a comment.